lang/py/src/avro/schema.py

```diff
@@ -551,6 +506,69 @@ def to_json(self, names=None):
   def __eq__(self, that):
     return self.props == that.props
 
+#
+# Decimal Fixed Type
+#
+
+class FixedDecimalSchema(FixedSchema, LogicalSchema):
+  def __init__(self, schema_type, size, name, precision, scale=0, namespace=None, names=None, other_props=None):
+    if schema_type != 'fixed':
+      raise SchemaParseException('Logical type DECIMAL must be backed by fixed or bytes.')
+    if size < 0:
+      raise SchemaParseException('Size of a fixed length should be greater than 0')
+
+    self._validate({'max_precision': self._max_precision(size),
+                   'precision': precision,
+                   'scale': scale,
+                   'size': size})
+
+    LogicalSchema.__init__(self, 'decimal')
+    FixedSchema.__init__(self, name, namespace, size, names, other_props)
+    self.set_prop('precision', precision)
+    self.set_prop('scale', scale)
+
+  def _validate(self, validation_props=None):
```

Comment:
Two things:
1. If we consolidate the validation now and the specs change later then we may need to refactor again.
2. This file doesn't seem to contain any methods except for the two public APIs so I didn't want to modify the structure.